cmd/contour/serve.go

```diff
@@ -405,9 +425,12 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 	return g.Run()
 }
 
-func registerEventHandler(informers []cache.SharedIndexInformer, inf cache.SharedIndexInformer, handler cache.ResourceEventHandler) []cache.SharedIndexInformer {
-	inf.AddEventHandler(handler)
-	return append(informers, inf)
+func registerEventHandler(informers []cache.SharedIndexInformer, handler cache.ResourceEventHandler, addinformers ...cache.SharedIndexInformer) []cache.SharedIndexInformer {
```

Comment:
I'm not sure about this, there don't seem to be many callers who are registering more than one informer.

tbh, the signature of this method just went off the deep end, I think it needs to be rethought.

Reply:
Yeah, I'll back this change out, and log an issue about registerEventHandler.